lib/adapters/http.js

```diff
@@ -250,16 +254,22 @@ module.exports = function httpAdapter(config) {
     });
 
     // Handle request timeout
+    //
+    // There are many types of timeouts:
+    // DNS Timeout, Connect Timeout, Read/Write Timeout (idle timeout), and Call Timeout.
+    // Call Timeout measures from DNS Lookup to response body receiving completion.
+    //
+    // Currently, `timeout` option in Browser XHR indicates "Call Timeout".
+    // Create internal timer to match browser-side behavior.
     if (config.timeout) {
-      // Sometime, the response will be very slow, and does not respond, the connect event will be block by event loop system.
-      // And timer callback will be fired, and abort() will be invoked before connection, then get "socket hang up" and code ECONNRESET.
-      // At this time, if we have a large number of request, nodejs will hang up some socket on background. and the number will up and up.
-      // And then these socket which be hang up will devoring CPU little by little.
-      // ClientRequest.setTimeout will be fired on the specify milliseconds, and can make sure that abort() will be fired after connect.
-      req.setTimeout(config.timeout, function handleRequestTimeout() {
+      // Set internal timer to detect "Call Timeout"
+      // @see https://github.com/axios/axios/issues/2710#issuecomment-605693456
+      timer = setTimeout(function handleReadTimeout() {
+        // Stop receiving remaining bytes and close socket
         req.abort();
-        reject(createError('timeout of ' + config.timeout + 'ms exceeded', config, 'ECONNABORTED', req));
-      });
+
+        reject(createError('timeout of ' + config.timeout + 'ms exceeded', config, 'ETIMEDOUT', req));
```

Comment:
ECONNABORTED is changed as ETIMEDOUT. Do you share the same idea with #1543?

Reply:
The ETIMEOUT on server side different with browser. the http request on nodejs while be block the js until the poll finished (successed or failed),. so the settimeout can not be executed when connection failed.

you can read the detail from here : https://nodejs.org/en/docs/guides/event-loop-timers-and-nexttick/

Reply:
Actually, it was a unintended change. Sorry about that.
Anyway, Yes. I think it should be a `ETIMEDOUT`. 

As a axios consumer, There's no reliable ways to know why given request was timed out or not. Current error does not help consumers to detect timeouts easily. Currently, It only can be done by by matching error message like below:

```typescript
const isTimeoutError = /timed?out/i.test(error.message);
// or
const isTimeoutError = error.message("timeout");
```

I think it is not reliable way to match timeout errors. 

Another reason why we should change error code is for clarity. `ETIMEDOUT` error code is more clearer than `ECONNABORTED`. because timeout was really happened so that's why we aborted request. I also think `ECONNABORTED` is not actual root cause of given error.

Final reason why we should change error code is for consistency. Node.js built-in modules use `ETIMEDOUT` error code for timeout errors.

Reply:
Do we think this would be a breaking change tho? Current users of axios might expect it to be `ECONNABORTED` and might rely on that for their own implementations.

Reply:
Yeah, It would be a breaking change. If users only relied to `ECONNABORTED` error code. it would break their applications.

Reply:
@chinesedfan Any plans to start using [semver](https://semver.org) based versioning?

Reply:
@fishcharlie Yes, next version should be 0.20 at least, which means breaking change because it is less than 1.0. And 0.19.1 is not a standard version.